ceph/ceph_admin/host_mixin.py

```diff
@@ -0,0 +1,262 @@
+                    remote=self.installer,
+                    args=[
+                        "ceph",
+                        "orch",
+                        "host",
+                        "label",
+                        "add",
+                        node.shortname,
+                        label,
+                    ],
+                )
+                assert label in self.fetch_labels_by_hostname(node.shortname)
+
+    def remove_label(self, **config):
+        """
+        Removes label from nodes
+
+        - remove labels from existing nodes
+        - if nodes are empty, all cluster nodes are considered
+
+        config:
+            nodes: [node1.obj, node2.obj, ...]
+
+        Args:
+            config
+        """
+        nodes = config.get("nodes", self.cluster.get_nodes())
+        nodes = nodes if isinstance(nodes, list) else [nodes]
+        logger.info(
+            "Remove label on this nodes : %s" % [node.ip_address for node in nodes]
+        )
+
+        for node in nodes:
+            if node.shortname not in self.fetch_host_names():
+                continue
+
+            labels = node.role.role_list
+            for label in labels:
+                self.shell(
+                    remote=self.installer,
+                    args=["ceph", "orch", "host", "label", "rm", node.shortname, label],
+                )
+                # assert label not in self.fetch_labels_by_hostname(node.shortname)
+
+    def set_address(self, **config):
+        """
+        Set Ip address to node
+
+        - Attach labels to existing nodes
+        - if nodes are empty, all cluster nodes are considered
+        - roles defined to each node used as labels( eg., [mon, mgr])
+
+        config:
+            nodes: [node1.obj, node2.obj, ...]
+
+        Args:
+            config
+        """
+        nodes = config.get("nodes", self.cluster.get_nodes())
+        nodes = nodes if isinstance(nodes, list) else [nodes]
+        logger.info(
+            "Set Address on this nodes : %s" % [node.ip_address for node in nodes]
+        )
+        for node in nodes:
+            if node.shortname not in self.fetch_host_names():
+                # Add host
+                self.host_add(node)
+
+            self.shell(
+                remote=self.installer,
+                args=[
+                    "ceph",
+                    "orch",
+                    "host",
+                    "set-addr",
+                    node.shortname,
+                    node.ip_address,
+                ],
+            )
+            assert node.ip_address in self.get_addr_by_name(node.shortname)
+
+    def fetch_labels_by_hostname(self, node_name):
+        """
+        Fetch labels attach to a node by hostname
+        Returns:
+            labels: list of labels
+        """
+        nodes = self.host_list()
+        for node in nodes:
+            if node_name in node["hostname"]:
+                return node["labels"]
+        return []
```

Comment:
Would it be a good idea to raise an exception `raise ResourceNotFound`?

Reply:
Currently used to validate label in another method, would keep it to next improvements.